### PR TITLE
Consider the case where the path points at a directory.

### LINKED ISSUE #195: When it tries to checkout, it doesn't know how to delete directories
Check https://github.com/PressLabs/gitfs/blob/master/gitfs/repository.py#L92

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/24261266-when-it-tries-to-checkout-it-doesn-t-know-how-to-delete-directories?utm_campaign=plugin&utm_content=tracker%2F7081451&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F7081451&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by bogdanpetrea**
Fix #195

### CODE REVIEW

File: gitfs/repository.py [hunk 1]
```diff
@@ -89,17 +94,23 @@ def checkout(self, ref, *args, **kwargs):
...
+            full_path = self._full_path(path)
             if path not in self._repo.index:
                 if path not in self.ignore:
-                    os.unlink(self._full_path(path))
+                    try:
+                        os.unlink(full_path)
+                    except OSError:
+                        # path points to a directory containing untracked files
+                        rmtree(full_path,
+                               onerror=Repository.checkout_fail_rmtree_callback)
```
Comment (vtemian) [inline gitfs/repository.py]: You can use a lambda here instead of a static method

**Comment by vtemian** [comment]
Can you please write a test for this use case?

File: gitfs/repository.py (hunk 1 above)
Comment (krodyrobi) [inline gitfs/repository.py]: :+1:

Comment (bogdanpetrea) [inline gitfs/repository.py]: Good idea!

**Comment by vtemian** [comment]
Tests are still failing

**Pull request merged**